board/xen/xenguest_arm64/xenguest_arm64.c

```diff
@@ -63,6 +66,90 @@ static int get_next_memory_node(const void *blob, int mem)
 	return mem;
 }
 
+#ifdef CONFIG_VIRTIO_BLK
+#ifdef CONFIG_VIRTIO_PCI
+static void add_pci_mem_map(const void* blob, int* cnt)
+{
+	struct fdt_resource reg_res;
+	int node = -1, len = 0, cells_per_record = 0, max_regions = 0;
+	int pci_addr_cells = 0, addr_cells = 0, size_cells = 0;
+
+	if ((*cnt) >= MAX_MEM_MAP_REGIONS)
+		return;
+	node = fdt_node_offset_by_prop_value(blob, -1,"compatible", "pci-host-ecam-generic", 22);
+	if (node == -FDT_ERR_NOTFOUND)
+		return;
+	if (fdt_get_resource(blob, node, "reg", 0, &reg_res) == -FDT_ERR_NOTFOUND)
+		return;
+	xen_mem_map[*cnt].virt = reg_res.start;
+	xen_mem_map[*cnt].phys = reg_res.start;
+	xen_mem_map[*cnt].size = fdt_resource_size(&reg_res);
+	xen_mem_map[*cnt].attrs = (PTE_BLOCK_MEMTYPE(MT_NORMAL) |
+				PTE_BLOCK_INNER_SHARE);
+	(*cnt)++;
+
+	const u32 *prop = fdt_getprop(blob,node, "ranges", &len);
+	if (!prop)
+		return;
+
+	pci_addr_cells =  fdt_address_cells(blob, node);
+	addr_cells = fdt_address_cells(blob, 0);
+	size_cells = fdt_size_cells(blob, node);
+
+	/* PCI addresses are always 3-cells */
```

Comment:
Is this really true?

Reply:
This is original comment, from pci code, from pci-uclass.c

Reply:
"The numerical representation of a PCI
address consists of three cells, encoded as follows" https://www.devicetree.org/open-firmware/bindings/pci/pci2_1.pdf